File: ARCADE/The Core/Level 4_9: candles.cs
    using System;
     
    public class Test
    {
    	public static int candles(int candlesNumber, int makeNew){
            int result = candlesNumber, sur = 0;
            while (candlesNumber != 0){
                sur = candlesNumber % makeNew + sur;
                result = result + candlesNumber / makeNew;
                if (sur / makeNew != 0){
                    result = result + sur / makeNew;
                    sur = sur % makeNew;
                }
                candlesNumber = candlesNumber / makeNew;
            }
            return result;
        }
    	public static void Main()
    	{
    		int candlesNumber = 5, makeNew = 2;
    		Console.Write(candles(candlesNumber, makeNew));
    	}
    }

File: ARCADE/The Core/Level 4_9: candles.pas
program candles;

function candles(candlesNumber: integer; makeNew: integer): integer;
var rs, sur : integer;
begin
    rs := candlesNumber; sur := 0;
    while(candlesNumber <> 0) do
    begin
        sur := candlesNumber mod makeNew + sur;
        inc(rs, candlesNumber div makeNew);
        if(sur div makeNew <> 0) then
        begin
            inc(rs, sur div makeNew);
            sur := sur mod makeNew;
        end;
        candlesNumber := candlesNumber div makeNew;
    end;
    
    candles := rs;
end;

var candlesNumber, makeNew : integer;
begin
	candlesNumber := 5; makeNew := 2;
	writeln(candles(candlesNumber, makeNew));
end.
